hardware_interface/src/resource_manager.cpp

```diff
@@ -91,6 +91,22 @@ std::string interfaces_to_string(
   return ss.str();
 };
 
+void get_hardware_related_interfaces(
+  const std::vector<std::string> & hw_command_itfs,
+  const std::vector<std::string> & start_stop_interfaces_list,
+  std::vector<std::string> & hw_interfaces)
```

Comment:
I know this is here for a specific reason but maybe we could still distill it down to it's core concept
```suggestion
void intersect(
  const std::vector<std::string> & larger_list,
  const std::vector<std::string> & shorter_list,
  std::vector<std::string> & intersection)
```